### ISSUE TITLE
💅 Improve `nursery/noNodejsModules` detection (ignore type only imports and apply it only on `use-client` directives)

### ISSUE THREAD

**Issue by kevinwolfcr** [biomejs/biome#1674]
### Environment information

```bash
CLI:
  Version:                      1.5.3
  Color support:                true

Platform:
  CPU Architecture:             aarch64
  OS:                           macos

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.0.20"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### Rule name

`nursery/noNodejsModules`

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=aQBtAHAAbwByAHQAIAB0AHkAcABlACAAewAgAEEAcwB5AG4AYwBMAG8AYwBhAGwAUwB0AG8AcgBhAGcAZQAgAH0AIABmAHIAbwBtACAAIgBuAG8AZABlADoAYQBzAHkAbgBjAF8AaABvAG8AawBzACIAOwAKAAoAZQB4AHAAbwByAHQAIAB0AHkAcABlACAAVABlAHMAdAAgAD0AIABBAHMAeQBuAGMATABvAGMAYQBsAFMAdABvAHIAYQBnAGUA

### Expected result

1. On the attached link, I am only importing a type from a Node module, should it report?
2. A good option to make sure the file includes client-only code is to check if the `use client` directive is being used.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by anonrig** [comment]
Makes sense to me. I'll update the implementation.

**Comment by Conaclos** [comment]
> A good option to make sure the file includes client-only code is to check if the use client directive is being used.

Should we check this? If we check that we could make the rule recommended?

**Comment by ematipico** [comment]
> > A good option to make sure the file includes client-only code is to check if the use client directive is being used.
> 
> Should we check this? If we check that we could make the rule recommended?

Nowadays it isn't about client anymore, but runtime too. Some code doesn't necessarily run on a client, but it could run on Deno, Bun, Cloudflare workers and more. So no, we can't rely on a directive, and we can't make the rule recommended.
